### ISSUE TITLE
Running on an Apple M1/M2 chip?

### ISSUE THREAD

**Issue by ben-kolber** [serp-ai/bark-with-voice-clone#52]
Hey, was wondering if anyone tried running the model on a macbook pro GPU using the new chips.

**Comment by keithcarmona** [comment]
To get the voice cloning to work on an M1, you might have to modify the customtokenizer.py file and where it says,model.load_state_dict(torch.load(path))  ... change it to: model.load_state_dict(torch.load(path, map_location=torch.device('cpu'))).   . Make a backup of the file before to modify anything.

**Comment by ben-kolber** [comment]
Thanks! @keithcarmona 

I was also wondering whether that was the file to modify in order to have the model run on 2 GPU's using the Pytorch torch.nn.DataParallel() function. 

Does the model run faster on 2 GPU's? It's usually size dependent was wondering if you knew anything about that. Thanks :)

**Comment by keithcarmona** [comment]
I'm not sure about the coding changes required to use two GPU's. To get the cloning working on an M1 I used the code as a quick fix, but I switched it back to the original file afterwards because I found the updated code made things slower. I only had to clone one voice, so I figured it was probably wait for an official update before I started changing too many things.

**Comment by JulienVincenot** [comment]
> To get the voice cloning to work on an M1, you might have to modify the customtokenizer.py file and where it says,model.load_state_dict(torch.load(path)) ... change it to: model.load_state_dict(torch.load(path, map_location=torch.device('cpu'))). . Make a backup of the file before to modify anything.

Hi, not sure if I understand correctly is this fix only making it work on macOS but with CPU (slow) or is it only a part of the code? There is no way to run with Metal using last versions of PyTorch nightly? I just got tortoise tts to work on a M2 with GPU but it seems to support only English... :( 

(sorry that's probably a stupid question, i'm a total noob to this)

**Comment by keithcarmona** [comment]
I'm a total noob at this too. This was the only way I could get the voice cloning run on my M1 Max. The regular text to voice is very slow on my system and it takes around 3 minutes for a 24 second clip. It allows me to append with -enablemps at the end of the launch command, but I think it's only using the CPU. I'll probably mess around a little more. I actually just got GPT 4 to correct the error I was receiving when I attempted to launch. 

I haven't messed around with Bark in a few weeks because I couldn't get the voice cloning to work well. I had to use an external hugging face site to do the voice cloning with limited success.

**Comment by JulienVincenot** [comment]
Hi Keith thanks for your reply ! I'm slow to get back to you I was away without computer.
> I'm a total noob at this too. This was the only way I could get the voice cloning run on my M1 Max. The regular text to voice is very slow on my system and it takes around 3 minutes for a 24 second clip. 

Hm on a M1 Max I don't know, I have an entry level M2 macMini and with tortoise-tts I got to generate only a few seconds of text, each time in 3 different versions (i.e. audio files) for around that time 2-3 min. I'm sure the models are not comparable but wondering, when you ran bark on your Machine the GPU didn't show any activity, i.e. in GPU window of Activity Monitor?
For me that one was going full on when running tortoise.
If they both use PyTorch nightly I'm not sure why one would work with GPU and not the other, would need to dive deeper in the code but my Python is very rusty, I'm a musician so first trying to play around with some TTS and cloning tools to see if I can use them somehow for projects.

**Comment by polarNerd204** [comment]
Parallel note here….i had tried running the custom node (MARS5 and the UVR5 Node ) for comfyUI to do voice clone and TTS and STS. Got it all installed on my Mac Studio M2 Ultra and tried to test a workflow but hit the cuda error barrier. Wonder if changing that could would make it work ?
Thoughts?

**Comment by devinschumacher** [comment]
@ben-kolber 

We just launched our follow up to BARK -- an AI Voice Cloner Desktop App (runs locally, offline, super realistic voice cloning, data privacy, no coding needed)

👉 Check it out: https://github.com/serpapps/ai-voice-cloner

**Comment by platform-kit** [comment]
@devinschumacher that link and everything associated with Serp looks an awful lot like a spam/phishing attack. Why the scummy website? Do you have a legitimate explanation? (I very much doubt it)

**Comment by devinschumacher** [comment]
@platform-kit I totally agree with you. It's just a stupid store template from the CRM platform we're using (gohighlevel)

We just got on it (like two weeks ago) so i've been setting everything up but .... one thing at a time i guess. it's just me doing the frontend and i'm not a frontend guy. 

but one of my priorities is getting rid of that crap and building out something simple simpler and nice ! 

Definitely intend on doing that in the next week or so
